apps/web/src/contexts/payment-context.tsx

```diff
@@ -108,6 +131,29 @@ export function usePaymentProvider({
     [t]
   )
   const [formErrors, setFormErrors] = useState<FormValidation>()
+  const price =
+    release?.type === PackType.Auction
+      ? bid
+      : formatIntToFloat(release?.price || 0)
+
+  const handleSetStatus = useCallback(
+    (status: CheckoutStatus.form | CheckoutStatus.summary) => {
+      setStatus(status)
+      const step = status === CheckoutStatus.form ? 'details' : 'summary'
+      const path = `${asPath.split('?')[0]}?step=${step}`
+      if (path !== asPath) {
+        return push(`${asPath.split('?')[0]}?step=${step}`)
+      }
+      return
+    },
+    [asPath, push]
+  )
```

Comment:
Change both the status and the path to reflect accurately, which can be used in order to return to the form page after an error is found post-submission.